include/antler/project/project.hpp

```diff
@@ -49,132 +63,162 @@ class project {
-         return upsert_test(std::move(obj));
+      //else if constexpr (Ty == object::type_t::test)
+      //   return upsert_test(std::move(obj));
       else
          throw std::runtime_error("internal failure");
    }
 
    /// update or insert a new application object. If the object already exists, an update is performed by removing the old one and
    /// adding the new one.
    /// @param obj  The object to update or insert.
-   void upsert_app(object&& app) noexcept;
+   void upsert(app_t&& app) noexcept { m_apps[app.name()] = std::move(app); }
+
    /// update or insert a new library object. If the object already exists, an update is performed by removing the old one and
    /// adding the new one.
    /// @param obj  The object to update or insert.
-   void upsert_lib(object&& lib) noexcept;
+   void upsert(lib_t&& lib) noexcept { m_libs[lib.name()] = std::move(lib); }
+
+#if 0
    /// update or insert a new test object. If the object already exists, an update is performed by removing the old one and
    /// adding the new one.
    /// @param obj  The object to update or insert.
    void upsert_test(object&& test) noexcept;
+#endif
 
-   /// Search the lists to see if an object exists.
+   /// Does the given app exist in the project.
+   /// @param name  The app name to search for.
+   /// @return true if the app exists, false otherwise.
+   [[nodiscard]] inline bool app_exists(const std::string& name) const noexcept { return m_apps.count(name) > 0; }
+
+   /// Does the given lib exist in the project.
+   /// @param name  The lib name to search for.
+   /// @return true if the lib exists, false otherwise.
+   [[nodiscard]] inline bool lib_exists(const std::string& name) const noexcept { return m_libs.count(name) > 0; }
+
+   /// Does the given object exist in the project.
    /// @param name  The object name to search for.
-   /// @param type  If type is other than any, the search is limited to that single type.
-   /// @return true if an object with the provided name exists in the indicated list.
-   [[nodiscard]] bool object_exists(std::string_view name, object::type_t type = object::type_t::any) const noexcept;
+   /// @return true if the object exists, false otherwise.
+   [[nodiscard]] inline bool exists(const std::string& name) const noexcept { return app_exists(name) || lib_exists(name); }
+
 
    /// Return the first object with the matching name where search order is apps, libs, tests.
    /// @TODO replace this with a std::vector<antler::project::object>/antler::project::object::list_t to return all the objects
    /// with matching names.
    /// @param name  The name to search for in the object lists.
    /// @param type  If type is other than any, the search is limited to that single type.
    /// @return vector with copies of the objects.
-   [[nodiscard]] antler::project::object& object(std::string_view name); 
+   template <typename Tag>
+   [[nodiscard]] auto& object(const std::string& name) {
```

Comment:
RN, the doxygen comment references tests, but the code doesn't follow. Yet. Plans?